byway/src/main/webapp/home.js

```diff
@@ -6,11 +17,107 @@ else{
 }
 
 function initializeHomePage(){
-  // TODO(renau06): fetch /api/gettrips and display the users past trips
+  setLogoutLink();
+  createPastTrip();
   document.getElementById('create-trip').addEventListener('click', () => {
     fetch('/api/createtrip', {method: 'POST'}).then((response) => response.json()).then((trip) =>{
-      window.location.href = '/destinations.html?tripKey=' + trip.keyString;  
+      let tripKey = trip.keyString;
+      window.location.href = configureTripKeyForPath(tripKey, "/destinations.html");
     });
   });
 }
 
+function createPastTrip(){
+  fetch('/api/gettrips').then((response) => response.json()).then((tripIds) => {
+    let tripNum = 1;
+    tripIds.forEach(trip => {
+      let isDestinationsMissing = trip.destinations.length == 0;
+      let isInterestsMissing = trip.interests.length == 0;
+      if(isDestinationsMissing|| isInterestsMissing) { 
```

Comment:
Seems like you could just do this check inline rather than use two variables. In general, if you have a boolean value that is only read once, you can just get its value where it is used.

Reply:
I passed these variables into the showIncompleteTrip so i kept them as variables, but I changed to an inline check

Reply:
If they are used elsewhere you can ignore my comment. Either way is fine.